QFI.DigitalIO/Factories/BitControllerFactory.cs

```
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security;
+using System.Text;
+using System.Threading.Tasks;
+using QFI.DigitalIO.BitControllers;
+
+namespace QFI.DigitalIO.Factories
+{
+    public class BitControllerFactory : IBitControllerFactory
+    {
+        Dictionary<string, IBitController> _dictionary = new Dictionary<string, IBitController>();
+        public IBitController Create(DeviceModel deviceModel, string uniqueName)
+        {
+            if (BitControllerExist(uniqueName))
+                return _dictionary[uniqueName];
+            else
+            {
+                IBitController retBitController = null;
+
+                switch (deviceModel)
+                {
+                    case DeviceModel.USB1208LS:
+                        retBitController = new DaqBitController();
+                        break;
+                    case DeviceModel.MockDevice:
+                        retBitController = new MockBitController();
+                        break;
+                    default:
+                        retBitController = new MockBitController();
```

You should throw either an Argument exception or an Invalid operation exception here rather than returning MockBitController.

---

It is also common to use a Null object pattern if an exception is not suitable